packages/flutter/lib/src/widgets/sliver_prototype_item_list.dart

```diff
@@ -0,0 +1,188 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'sliver.dart';
+
+export 'package:flutter/rendering.dart' show RenderSliverFixedExtentBoxAdaptor;
+
+class SliverPrototypeExtentListElement extends SliverMultiBoxAdaptorElement {
+  SliverPrototypeExtentListElement(SliverPrototypeExtentList widget) : super(widget);
+
+  @override
+  SliverPrototypeExtentList get widget => super.widget;
+
+  @override
+  RenderSliverPrototypeExtentList get renderObject => super.renderObject;
+
+  Element _prototype;
+  static final Object _prototypeSlot = new Object();
+
+  @override
+  void insertChildRenderObject(covariant RenderObject child, covariant dynamic slot) {
+    if (slot == _prototypeSlot) {
+      assert(child is RenderBox);
+      renderObject.child = child;
+    } else {
+      super.insertChildRenderObject(child, slot);
+    }
+  }
+
+  @override
+  void didAdoptChild(RenderBox child) {
+    if (child != renderObject.child)
+      super.didAdoptChild(child);
+  }
+
+  @override
+  void moveChildRenderObject(RenderBox child, dynamic slot) {
+    if (slot == _prototypeSlot) {
+      renderObject.remove(child);
+      assert(child is RenderBox);
+      renderObject.child = child;
```

Comment:
I don't think this'll work for the case of moving a child from the list to the prototype entry.